### ISSUE TITLE
focusmaster dispatcher causes crash if there is no master 

### ISSUE THREAD

**Issue by proycon** [hyprwm/Hyprland#951]
This concerns the new feature implemented in #941 , which is exactly what I need. But when there is no master, it causes Hyprland to crash. I would have expected it to simply do nothing in this case (since there is nothing to focus on an empty workspace).

**Context**:

I implemented [a handler script](https://github.com/proycon/dotfiles/blob/579428b814a3d4b6486ebae687ef590b01975650/handlehyprland.sh#L11) to focus the master window whenever a window closes, works fine except when the master window itself closes... boom..               

(by the way, I wonder if such `focusmasteronclose` functionality might perhaps be desirable to implement in Hyprland itself using a boolean setting)

**Version:** Running on the git d0e47d9 .

**Relevant log portion**:

```
[LOG] Hyprctl: dispatcher layoutmsg : focusmaster
[WARN] Disabling TextInput on no IME!
[LOG] Window 745aa190 unmapped (class foot)
[LOG] Unregistered late callbacks XDG
[LOG] Callback 745aa1b8 -> 745aa1b0, XDG Window Late removed.
[LOG] Callback 745aa358 -> 745aa350, XDG Window Late removed.
[LOG] Callback 745aa490 -> 745aa488, XDG Window Late removed.
[LOG] Callback 745aa5c8 -> 745aa5c0, XDG Window Late removed.
[LOG] Callback 745aa560 -> 745aa558, XDG Window Late removed.
[LOG] Callback 745aa4f8 -> 745aa4f0, XDG Window Late removed.
[LOG] Callback 745aa630 -> 745aa628, XDG Window Late removed.
[LOG] Callback 745aa428 -> 745aa420, XDG Window Late removed.
[LOG] Framebuffer created, status 36053
[LOG] On closed window, new focused candidate is 0
[LOG] Destroying the SubSurface tree of unmapped window 745aa190
[LOG] Callback 745a7730 -> 745a7728, Subsurface removed.
[LOG] Callback 745a7660 -> 745a7658, Subsurface removed.
[LOG] Callback 745a76c8 -> 745a76c0, Subsurface removed.
[LOG] Callback 745a65d8 -> 745a65d0, SurfaceTreeNode removed.
[LOG] Callback 745a6640 -> 745a6638, SurfaceTreeNode removed.
[LOG] Callback 745a6570 -> 745a6568, SurfaceTreeNode removed.
[LOG] SurfaceTree Node removed
[LOG] Callback 745aa7d0 -> 745aa7c8, Toplevel removed.
[LOG] Callback 745aa768 -> 745aa760, Toplevel removed.
[LOG] Callback 745aa838 -> 745aa830, Toplevel removed.
[ERR] Error in getAppIDClass: basic_string: construction from null is not valid
[LOG] Window 745aa190 destroyed, queueing. (class )
[LOG] Callback 745aa220 -> 745aa218, XDG Window removed.
[LOG] Callback 745aa288 -> 745aa280, XDG Window removed.
[LOG] Callback 745aa2f0 -> 745aa2e8, XDG Window removed.
```

I hope this gives sufficient data to pinpoint the problem.

**Comment by proycon** [comment]
I forgot the instructions to reproduce the problem (without my script), just open a terminal and run:

```
$ (sleep 10 && hyprctl dispatch layoutmsg focusmaster) &
$ disown
$ exit
```

**Comment by vaxerski** [comment]
it concerned a bunch of others too. Fixed in 349afa0e7a5a8f60404772ae76ab0dc6fe4117ff
